doc/design/diagrams/logging/logging_classes.puml

```diff
@@ -0,0 +1,85 @@
+@startuml
+
+package platform_building_blocks {
+
+enum LogLevel {
+    OFF = 0,
+    FATAL,
+    ERROR,
+    WARN,
+    INFO,
+    DEBUG,
+    TRACE,
+}
```

Comment:
This is the same ordering like in the current implementation. Shall we follow swap the order to
```cpp
enum LogLevel {
    TRACE = 0,
    DEBUG,
    INFO,
    WARN,
    ERROR,
    FATAL,
    OFF,
}
```

Reply:
Another consideration. What do you think to rename `OFF` to `NONE`/`QUIET`/`SILENT` or something similar. `OFF` kind of does not fit with the other levels.

Reply:
* `SILENT` & `QUIET` is like I will log to a file but you won't see it
* `NONE` feels right here

One question, is it possible for the user to log a message to the `NONE` LogLevel so something like `Log(LogLevel::NONE) << "hello";` ? Or do we prevent this by providing only `LogTrace()` - `LogFatal()`

Reply:
Whoopsi. In my proof of concept it is not caught but it should not be possible.